core/src/client/lwm2m_static.c

```diff
@@ -148,6 +151,30 @@ AwaError AwaStaticClient_SetBootstrapServerURI(AwaStaticClient * client, const c
     return result;
 }
 
+AwaError AwaStaticClient_SetFactoryBootstrapInformation(AwaStaticClient * client, const BootstrapInfo * factoryBootstrapInformation)
+{
+    AwaError result = AwaError_Unspecified;
+
+    if ((client != NULL) && (factoryBootstrapInformation != NULL))
```

Comment:
might be handy to split out the null checks, and provide error messages for each